File: app/src/main/java/ua/hospes/rtm/ui/race/detail/RaceItemDetailContract.java
package ua.hospes.rtm.ui.race.detail;

import java.util.List;

import ua.hospes.rtm.domain.race.models.RaceItem;
import ua.hospes.rtm.domain.sessions.models.Session;

/**
 * @author Andrew Khloponin
 */
interface RaceItemDetailContract {
    interface View {
        void onUpdateRaceItem(RaceItem raceItem);

        void onUpdateSessions(List<Session> sessions);
    }
}

File: app/src/main/java/ua/hospes/rtm/ui/race/detail/RaceItemDetailContract.kt
package ua.hospes.rtm.ui.race.detail

import ua.hospes.rtm.domain.race.models.RaceItem
import ua.hospes.rtm.domain.sessions.Session

interface RaceItemDetailContract {
    interface View {
        fun onRaceItem(item: RaceItem)
        fun onSessions(list: List<Session>)

        fun onError(t: Throwable)
    }
}